File: configmasterhelper/src/main/java/com/spascoding/configmasterhelper/javahelper/ConfigMasterHelper.java
package com.spascoding.configmasterhelper.javahelper;

import android.content.ContentValues;
import android.content.Context;
import android.database.Cursor;
import android.net.Uri;
import android.util.Log;

public class ConfigMasterHelper {

    private static final Uri CONFIG_URI =
            Uri.parse("content://com.spascoding.configmaster.data.provider.ConfigProvider/config");

    /**
     * Insert or update configuration into ConfigMaster
     */
    public static void insertConfig(Context context, String configName, String jsonData) {
        ContentValues values = new ContentValues();
        values.put("configName", configName);
        values.put("jsonData", jsonData);

        context.getContentResolver().insert(CONFIG_URI, values);
    }

    /**
     * Fetch configuration from ConfigMaster
     */
    public static String fetchConfig(Context context, String configName) {
        Cursor cursor = context.getContentResolver().query(
                CONFIG_URI,
                null,
                null,
                new String[]{configName},
                null
        );

        String jsonData = null;
        if (cursor != null) {
            try {
                if (cursor.moveToFirst()) {
                    jsonData = cursor.getString(cursor.getColumnIndexOrThrow("jsonData"));
                    Log.d("ConfigMasterHelper", "Fetched config = " + jsonData);
                }
            } finally {
                cursor.close();
            }
        }
        return jsonData;
    }
}

File: configmasterhelper/src/main/java/com/spascoding/configmasterhelper/kotlinhelper/ConfigMasterHelper.kt
package com.spascoding.configmasterhelper.kotlinhelper

import android.content.ContentValues
import android.content.Context
import android.net.Uri
import android.util.Log

object ConfigMasterHelper {

    private val CONFIG_URI: Uri =
        Uri.parse("content://com.spascoding.configmaster.data.provider.ConfigProvider/config")

    /**
     * Insert or update configuration into ConfigMaster
     */
    fun insertConfig(context: Context, configName: String, jsonData: String) {
        val values = ContentValues().apply {
            put("configName", configName)
            put("jsonData", jsonData)
        }
        context.contentResolver.insert(CONFIG_URI, values)
    }

    /**
     * Fetch configuration from ConfigMaster
     */
    fun fetchConfig(context: Context, configName: String): String? {
        val cursor = context.contentResolver.query(
            CONFIG_URI,
            null,
            null,
            arrayOf(configName),
            null
        )

        var jsonData: String? = null
        cursor?.use {
            if (it.moveToFirst()) {
                jsonData = it.getString(it.getColumnIndexOrThrow("jsonData"))
                Log.d("ConfigMasterHelper", "Fetched config = $jsonData")
            }
        }
        return jsonData
    }
}
